tests/foreman/api/test_classparameters.py

```diff
@@ -542,14 +818,20 @@ def test_negative_validate_matcher_non_existing_attribute(self):
         2. Create a matcher with non existing attribute in org.
 
         @assert: Error raised for non existing attribute.
-
-        @caseautomation: notautomated
-
-        @CaseLevel: System
         """
+        sc_param = self.sc_params_list.pop()
+        with self.assertRaises(HTTPError) as context:
+            entities.OverrideValue(
+                smart_class_parameter=sc_param,
+                match='non_existing_attribute=example.com',
```

Comment:
Wrong Implementation. Here non existing attribute can only be tested with Host/Hostgroup.

In short you choose match as match='fqdn=nonexisting.com' or 'hostgroup=nonexistingHG' where these host and hostgroup are not actually existing in satellite.